### DISCUSSION TITLE
Training models for multiple subclasses from a general class

### DISCUSSION THREAD

**Discussion by AdwaitMD** [ultralytics/ultralytics#13174] [Q&A]
Hi, 
I have a specific problem. First, I want to train the yolov8 pose model to detect small objects (e.g. goats) from 5K images. then, I want to use a custom tracker and track the objects across the entire video.  (I have already done this). In the next step, I want to extract crops of the different tracks/ objects, and train the yolov8-pose again to detect individual objects separately as different classes (e,g. goat 1, goat 2, goat 3) [All the objects are slightly different looking].  How can I do this? What is the best way to create training data out of crops?

**Comment by pderrenger** [comment]
Hi there,

Great to hear about your progress with YOLOv8! To achieve your goal of training the YOLOv8-pose model to detect individual objects as separate classes, you can follow these steps:

1. **Extract Crops**: Use the object cropping functionality to extract individual objects from your tracked video. You can refer to the [Object Cropping Guide](https://docs.ultralytics.com/guides/object-cropping/) for detailed instructions and code examples.

2. **Label the Crops**: Once you have the cropped images, manually label each crop with its respective class (e.g., goat 1, goat 2, goat 3). Tools like [LabelImg](https://github.com/tzutalin/labelImg) can be helpful for this task.

3. **Prepare the Dataset**: Organize your labeled crops into a dataset format compatible with YOLOv8. Ensure your dataset YAML file correctly references the paths to your images and annotations.

4. **Train the Model**: Use the YOLOv8 training script to train your model on the new dataset. You can start with a command like:
    ```bash
    yolo pose train data=your_dataset.yaml model=yolov8-pose.pt epochs=100 imgsz=640
    ```

By following these steps, you should be able to train your YOLOv8-pose model to recognize individual objects as distinct classes. If you need further assistance, feel free to ask! 😊

Best of luck with your project!

Kind
